src/python/pants/init/extension_api.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2025 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from collections.abc import Callable, Iterable
+from typing import Any, Protocol
+
+from pants.build_graph.build_file_aliases import BuildFileAliases
+from pants.engine.rules import Rule
+from pants.engine.target import Target
+from pants.engine.unions import UnionRule
+from pants.goal.auxiliary_goal import AuxiliaryGoal
+from pants.option.subsystem import Subsystem
+
+
+class ExtensionInitContextV0(Protocol):
```

Comment:
Advice on the best name for the context class?

Reply:
`PluginRegistrationContext`? We don't use "extension" anywhere with this meaning AFAICT.

Reply:
I got the "extension" name from the fact that the loader is in `extension_loader.py`. But I agree, we use the name "plugin" with our users. So this file should become `plugin_api.py`.

Should `extension_loader.py` be renamed to `plugin_loader.py`?

Also, should this file be moved to a more "official API" location (and not be in the `pants.init` package?

Reply:
Yeah, probably extension_loader.py should be renamed